### ISSUE TITLE
I can't monitor with IRPMon

### ISSUE THREAD

**Issue by Ata-Xan** [MartinDrab/IRPMon#8]
hi,
unfortunately even after i chose capture event IRPMon cant show the irp request and the display always white.... what should i do? and when i test the irpmonconsole --intall it shows an ERROR StartService():183

I appreciate your prompt reply
with best regards

**Comment by MartinDrab** [comment]
Did you selected drivers and/or devices you wish to monitor?

Irpmonconsole is actually quite obsolete. I used it mainly during theearly development stages. The error it is reporting means that the IRPMon driver is already running in the system, so the `--install` command is not necessary.

**Comment by Ata-Xan** [comment]
Thank you for your prompt response to the question, i just want to monitor changes that could happen for a file  (read,write,create,...) in a specific path ....on win 8 32 bit version but i have another 2 questions:
how could i add process name and the path to this? i mean the path that the file changes happened
I don't have any driver development experience my question is that is it possible to integrate this(IRPMon) with my java app you know i need to monitor IRP in my java app but i don't know how to do it, I would appreciate if you could give me some clues.

**Comment by MartinDrab** [comment]
Hello,

IRPMon is not suitable for monitoring file operations. Yes, it can monitor requests arriving at certain device(s), including devices representing file system instances. However, to monitor operations only on a specific location, you need to parse some of the requests to extract the file name (or even ask the file system for the name explicitly) and keep track of these information. IRPMon can't do this, since it does not know anything abou file systems and file operations. Of rouces, it is possible to add such a functionality but I am not planning to do so right now.

In short, IRPMon is not meant for such task. Even if you had experience in kernel development, you would probably do better to write a file system minifilter driver to serve your purposes than adopting IRPMon.

Look at the Process Monitor tool from Sysinternals, with propper filters, it should do exactly what you want.

Yes, you can communicate with a kernel driver also from a Java application, however, you would need to learn about native methods and Java Native Interface (JNI), and probably code a small library in C/C++ or other native language.

**Comment by Ata-Xan** [comment]
thank you very much for your very helpful comment.
